File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F14.P3.cpp
#include <cstdio>

#include "util.h"

double tri_area() {
// BEGIN PROBE F14.P3
    struct Tri : Shape {
        double b = 3.0, h = 4.0;
        double area() const override { return 0.5 * b * h; }
    };
    Tri t;
    const Shape& s = t;
    return s.area();
// END PROBE F14.P3
}

int main() {
    std::printf("%.6f\n", tri_area());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F14.P3.rs
mod shapes;

fn tri_area() -> f64 {
// BEGIN PROBE F14.P3
struct Tri { b: f64, h: f64 }
impl shapes::Shape for Tri {
    fn area(&self) -> f64 { 0.5 * self.b * self.h }
}
let s: &dyn shapes::Shape = &Tri { b: 3.0, h: 4.0 };
s.area()
// END PROBE F14.P3
}

fn main() {
    println!("{}", tri_area());
}
